Reorganised urEventGetProfilingInfo test into separate tests instead of a switch

martygrant opened the pull request:
For https://github.com/oneapi-src/unified-runtime/issues/2290

Also done a housekeeping pass across all urXGetInfo tests, renaming and adding consts where appropriate, and adding extra checks to tests with vectors.

RossBrunton commented on test/conformance/kernel/urKernelGetInfo.cpp:
```diff
@@ -12,100 +12,106 @@ using urKernelGetInfoTest = uur::urKernelTest;
   ASSERT_SUCCESS(urKernelGetInfo(kernel, property_name, property_size,
                                  property_value.data(), nullptr));
+
+  ASSERT_EQ(property_size, property_value.size());
```
NIT: Isn't this just testing that the standard library's std::vector class works rather than UR?

martygrant replied:
so my intention was to test that the entry point populates `pPropValue` and that its size matches the size retrieved from the first call

kbenzie replied:
Are you expecting the `std::vector`'s size to be changed by the call to `urKernelGetInfo()`? If so that's never going to happen, only the data owned by the `std::vector` will change not the container itself.

martygrant replied:
discussed this offline and cleared up my misunderstanding around testing for this, will update the PR with a better check

martygrant replied:
I've removed these size checks and added a helper function `uur::stringPropertyIsValid` to validate `char[]` returns

martygrant commented:
I've added some more changes to this PR, tidying up some more `char[]` return query tests, and added in a helper macro `ASSERT_GET_INFO` to validate numerical queries. Could use a re-review.
